File: trunk/ShadowFrameworkJS/SF20OpenGL20Graphics/src/shadow/pipeline/openGL20/tutorials/geometriesExample/LateralTube.js

function LateralTube(){
}

LateralTube.prototype = {

	generateGeometry:function(primitive){
		 SFCurvedTubeFunction  function = new  SFCurvedTubeFunction();
	//Primo Pezzo		function.Cc.add(new SFVertex3f(0,0,0));//Warning: Not well Identified 
		function.Cc.add(new SFVertex3f(0.0f,0.1f,0));
		function.Cc.add(new SFVertex3f(0.0f,0.2f,0));
		function.Cc.add(new SFVertex3f(0.0f,0.3f,0));
		function.Cc.add(new SFVertex3f(0.0f,0.4f,0));
		function.Rc.add(new SFVertex3f(0.1f,0,0));
		function.Rc.add(new SFVertex3f(0.2f,0.1f,0));
		function.Rc.add(new SFVertex3f(0.1f,0.2f,0));
		function.Rc.add(new SFVertex3f(0.1f,0.3f,0));
		function.Rc.add(new SFVertex3f(0.1f,0.4f,0));
	//Secondo Pezzo		function.Cc.add(new SFVertex3f(0.0f,0.5f,0));//Warning: Not well Identified 
		function.Cc.add(new SFVertex3f(0.1f,0.5f,0));
		function.Cc.add(new SFVertex3f(0.2f,0.5f,0));
		function.Rc.add(new SFVertex3f(0.1f,0.4f,0));
		function.Rc.add(new SFVertex3f(0.1f,0.4f,0));
		function.Rc.add(new SFVertex3f(0.2f,0.4f,0));
	//Terzo Pezzo		function.Cc.add(new SFVertex3f(0.3f,0.5f,0));//Warning: Not well Identified 
		function.Rc.add(new SFVertex3f(0.2f,0.4f,0));
		 SFConcreteTriangleExtractor  trianglesExtractor = new  SFConcreteTriangleExtractor();
		 SFQuadsSurfaceGeometry  quadsSurfaceGeometry = new  SFQuadsSurfaceGeometry(primitive,				function, new ,TexCoordFunction(), trianglesExtractor, 6, 8);
		quadsSurfaceGeometry.compile();
		return ,quadsSurfaceGeometry;
	}

};

File: ShadowFrameworkLite20_OpenGL20/src/shadow/pipeline/openGL20/tutorials/geometriesExample/LateralTube.java
package shadow.pipeline.openGL20.tutorials.geometriesExample;

import shadow.geometry.editing.SFConcreteTriangleExtractor;
import shadow.geometry.functions.SFCurvedTubeFunction;
import shadow.geometry.geometries.SFQuadsSurfaceGeometry;
import shadow.pipeline.SFPrimitive;
import shadow.system.data.objects.SFVertex3fData;

public class LateralTube {
	
	public SFQuadsSurfaceGeometry generateGeometry(SFPrimitive primitive) {
		
		SFCurvedTubeFunction function=new SFCurvedTubeFunction();

		//Primo Pezzo
		function.getData().getCc().add(new SFVertex3fData(0,0,0));
		function.getData().getCc().add(new SFVertex3fData(0.0f,0.1f,0));
		function.getData().getCc().add(new SFVertex3fData(0.0f,0.2f,0));
		function.getData().getCc().add(new SFVertex3fData(0.0f,0.3f,0));
		function.getData().getCc().add(new SFVertex3fData(0.0f,0.4f,0));

		function.getData().getRc().add(new SFVertex3fData(0.1f,0,0));
		function.getData().getRc().add(new SFVertex3fData(0.2f,0.1f,0));
		function.getData().getRc().add(new SFVertex3fData(0.1f,0.2f,0));
		function.getData().getRc().add(new SFVertex3fData(0.1f,0.3f,0));
		function.getData().getRc().add(new SFVertex3fData(0.1f,0.4f,0));

		//Secondo Pezzo
		function.getData().getCc().add(new SFVertex3fData(0.0f,0.5f,0));
		function.getData().getCc().add(new SFVertex3fData(0.1f,0.5f,0));
		function.getData().getCc().add(new SFVertex3fData(0.2f,0.5f,0));

		function.getData().getRc().add(new SFVertex3fData(0.1f,0.4f,0));
		function.getData().getRc().add(new SFVertex3fData(0.1f,0.4f,0));
		function.getData().getRc().add(new SFVertex3fData(0.2f,0.4f,0));
		
		//Terzo Pezzo
		function.getData().getCc().add(new SFVertex3fData(0.3f,0.5f,0));
		function.getData().getRc().add(new SFVertex3fData(0.2f,0.4f,0));
		
		SFConcreteTriangleExtractor trianglesExtractor=new SFConcreteTriangleExtractor();
		SFQuadsSurfaceGeometry quadsSurfaceGeometry=new SFQuadsSurfaceGeometry(primitive,
				function, new TexCoordFunction(), trianglesExtractor, 6, 8);
		
		quadsSurfaceGeometry.compile();
		return quadsSurfaceGeometry;
	}
}
